3rdparty/crypto/span.h

```diff
@@ -0,0 +1,378 @@
+// Copyright (c) 2018-2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_SPAN_H
```

Comment:
I'd rather not bring in this custom implementation of span if we can avoid it (instead we should use `std::span` once C++20 is widely available).

Can we convert `Write` and `Finalize` to use the same interface as the existing `CSHA256` class? That way we don't need this header.